guides/hotwire-turbo-micronaut-chat/hotwire-turbo-micronaut-chat.adoc

```diff
@@ -0,0 +1,740 @@
+----
+<!DOCTYPE html>
+<html lang="en" th:replace="~{layout :: layout(~{::script},~{::main})}" xmlns:th="http://www.thymeleaf.org">
+<head>
+    <script></script>
+</head>
+<body>
+    <main>
+        <h1 th:text="#{room.edit}"></h1>
+        <turbo-frame id="room">
+            <p th:replace="rooms/_edit :: edit(${room})"></p>
+        </turbo-frame>
+        <a th:href="@{|/rooms/${room.id}|}" th:text="#{action.show}"></a> |
+        <a href="/rooms" th:text="#{action.back}"></a>
+    </main>
+</body>
+</html>
+----
+
+
+We see our frame wrapped in blue.
+
+And when clicking the Edit link, the form from the Edit screen is presented.
+
+And upon submission, it's replaced again with just a display.
+
+If we go straight to the full page editing screen, we can see it has both a header and navigation links, parts we were emitting from the frame.
+
+=== Underscore Top
+
+Note that if we try to click a link within the frame that goes somewhere without a matching Frame, nothing happens.
+
+We can solve this by adding a Data Turbo Frame attribute that points to `_top` to break out of the frame, just like traditional HTML frames.
+
+Replace:
+
+[source, html]
+.src/main/resources/views/rooms/show.html
+----
+....
+<body>
+    <main>
+        ...
+        <turbo-frame id="room">
+            ...
+                <a href="/rooms" th:text="#{action.back}"></a>
+           </p>
+        </turbo-frame>
+....
+----
+
+with:
+
+[source, html]
+.src/main/resources/views/rooms/show.html
+----
+....
+<body>
+    <main>
+        ...
+        <turbo-frame id="room">
+            ...
+               <a data-turbo-frame="_top" href="/rooms" th:text="#{action.back}"></a>
+           </p>
+        </turbo-frame>
+....
+----
+
+Now the backlink works, and the frame scopes the edit display loop.
+
+=== Lazy Loading Frames
+
+Then, let's add the New Message link into an inline but lazy-loaded Turbo Frame tag that also, just for starters, acts on the whole page.
+
+This frame will be loaded right after the page displays, hitting the New Message Controller action we made earlier.
+
+Replace:
+
+[source, html]
+.src/main/resources/views/rooms/show.html
+----
+...
+...
+        <a href="/messages/create" th:text="#{message.new}"></a>
+    </main>
+</body>
+</html>
+----
+
+with:
+
+[source, html]
+.src/main/resources/views/rooms/show.html
+----
+....
+        <turbo-frame id="new_message"
+                     th:src="@{|/rooms/${room.id}/messages/create|}"
+                     target="_top"></turbo-frame>
+    </main>
+</body>
+</html>
+----
+
+==== Plug out the Frame
+
+Like with edit, we wrap the relevant segment in a Frame tag with a matching ID, which is how Turbo knows how to plug out the right frame.
+
+Replace:
+
+resource:views/messages/create.html[app=initial]
+
+with:
+
+resource:views/messages/create.html[app=complete]
+
+You can now see two requests when we load the room: one for the page, and one for the lazy-loader frame.
+
+Let's try to add a message.
```

Comment:
Do we need a browser screenshot around here?

Reply:
Plan is to record a screencast and put it embedded at the top of the guide.